pageserver/src/tenant/timeline.rs

```diff
@@ -389,6 +389,8 @@ pub struct Timeline {
     /// timeline is being deleted. If 'true', the timeline has already been deleted.
     pub delete_progress: Arc<tokio::sync::Mutex<DeleteTimelineFlow>>,
 
+    pub is_archived: AtomicBool,
```

Comment:
```suggestion
    pub(crate) is_archived: AtomicBool,
```

But is this needed at all? We could query the same from latest uploaded index part `RemoteTimelineClient::initialized_upload_queue()?.latest_uploaded_index()....` just as well.

Reply:
that would be less cheap though, ideally we use the `is_archived()` function in many places. if we had to take a lock, we'd also need to make `is_archived` an async fn (or tolerate blocking).

Reply:
removed the field now.

Reply:
Hmm now the is_archived is on RemoteTimelineClient, instead of it being on Timeline, which seems weird. Why that instead of using the suggested `RemoteTimelineClient::initialized_upload_queue()` from Timeline?

Reply:
It is there because it would allow something like future offloaded timelines could be something lighter weight. I was hopeful that `initialized_upload_queue` would had allowed to add these queries outside.